pkg/security/securitycontextconstraints/matcher.go

```diff
@@ -20,38 +21,58 @@ import (
 	"github.com/openshift/origin/pkg/security/uid"
 )
 
-// SCCMatcher defines interface for SecurityContextConstraint matcher
 type SCCMatcher interface {
-	FindApplicableSCCs(user user.Info) ([]*securityapi.SecurityContextConstraints, error)
+	FindApplicableSCCs(user user.Info, namespace string) ([]*securityapi.SecurityContextConstraints, error)
 }
 
-// DefaultSCCMatcher implements default implementation for SCCMatcher interface
-type DefaultSCCMatcher struct {
-	cache securitylisters.SecurityContextConstraintsLister
+type defaultSCCMatcher struct {
+	cache      securitylisters.SecurityContextConstraintsLister
+	authorizer authorizer.Authorizer
 }
 
-// NewDefaultSCCMatcher builds and initializes a DefaultSCCMatcher
-func NewDefaultSCCMatcher(c securitylisters.SecurityContextConstraintsLister) SCCMatcher {
-	return DefaultSCCMatcher{cache: c}
+func NewDefaultSCCMatcher(c securitylisters.SecurityContextConstraintsLister, authorizer authorizer.Authorizer) SCCMatcher {
+	return &defaultSCCMatcher{cache: c, authorizer: authorizer}
 }
 
-// FindApplicableSCCs implements SCCMatcher interface for DefaultSCCMatcher
-func (d DefaultSCCMatcher) FindApplicableSCCs(userInfo user.Info) ([]*securityapi.SecurityContextConstraints, error) {
+// FindApplicableSCCs implements SCCMatcher interface
+func (d *defaultSCCMatcher) FindApplicableSCCs(userInfo user.Info, namespace string) ([]*securityapi.SecurityContextConstraints, error) {
 	var matchedConstraints []*securityapi.SecurityContextConstraints
 	constraints, err := d.cache.List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 	for _, constraint := range constraints {
-		if ConstraintAppliesTo(constraint, userInfo) {
+		if ConstraintAppliesTo(constraint, userInfo) || d.authorizedForSCC(constraint, userInfo, namespace) {
```

Comment:
Updating ConstraintAppliesTo to take an authorizer and allowing nil (to support crazy CLI installer) will make the more obvious to people.